manifest.md

```diff
@@ -4,124 +4,100 @@ draft = true
-- `application/vnd.docker.distribution.manifest.v2+json`: New image manifest format (schemaVersion = 2)
-- `application/vnd.docker.distribution.manifest.list.v2+json`: Manifest list, aka "fat manifest"
-- `application/vnd.docker.image.rootfs.diff.tar.gzip`: "Layer", as a gzipped tar
-- `application/vnd.docker.container.image.v1+json`: Container config JSON
+- `application/vnd.oci.image.manifest.v1+json`: Image manifest format
+- `application/vnd.oci.image.manifest.list.v1+json`: Manifest list, aka "fat manifest"
+- `application/vnd.oci.image.serialization.v1+json`: Container config JSON
+- `application/vnd.oci.image.rootfs.tar.gzip`: "Layer", as a gzipped tar
 
 ## Manifest List
 
-The manifest list is the "fat manifest" which points to specific image manifests
-for one or more platforms. Its use is optional, and relatively few images will
-use one of these manifests. A client will distinguish a manifest list from an
-image manifest based on the Content-Type returned in the HTTP response.
+The manifest list is the "fat manifest" which points to specific image manifests for one or more platforms.
+Its use is optional, and relatively few images will use one of these manifests.
+A client will distinguish a manifest list from an image manifest based on the Content-Type returned in the HTTP response.
 
 ## *Manifest List* Field Descriptions
 
 - **`schemaVersion`** *int*
-	
-  This field specifies the image manifest schema version as an integer. This
-  schema uses the version `2`.
+
+  This REQUIRED property specifies the image manifest schema version.
+  This schema uses the version `2`.
 
 - **`mediaType`** *string*
 
-    The MIME type of the manifest list. This should be set to
-    `application/vnd.docker.distribution.manifest.list.v2+json`.
+  This REQUIRED property contains the MIME type of the referenced object.
+  This should be set to `application/vnd.oci.image.manifest.list.v1+json`.
 
 - **`manifests`** *array*
 
-    The manifests field contains a list of manifests for specific platforms.
+  This REQUIRED property contains a list of manifests for specific platforms.
+  While the property MUST be present, the size of the array MAY be zero.
 
-    Fields of a object in the manifests list are:
-    
-    - **`mediaType`** *string*
-    
-        The MIME type of the referenced object. This will generally be
-        `application/vnd.docker.image.manifest.v2+json`, but it could also
-        be `application/vnd.docker.image.manifest.v1+json` if the manifest
-        list references a legacy schema-1 manifest.
-    
-    - **`size`** *int*
-    
-        The size in bytes of the object. This field exists so that a client
-        will have an expected size for the content before validating. If the
-        length of the retrieved content does not match the specified length,
-        the content should not be trusted.
-    
-    - **`digest`** *string*
+  Fields of each object in the manifests list are:
```

Comment:
This is the confusion around having 0 mapping to even an OPTIONAL api, like
https://github.com/opencontainers/image-spec/issues/15

On Wed, Apr 13, 2016 at 11:03 AM, Doug Davis <notifications@github.com>
wrote:

> In manifest.md
> <https://github.com/opencontainers/image-spec/pull/12#discussion_r59564875>
> :
>
> > -    - **`mediaType`** *string*
> > -
> > -        The MIME type of the referenced object. This will generally be
> > -        `application/vnd.docker.image.manifest.v2+json`, but it could also
> > -        be `application/vnd.docker.image.manifest.v1+json` if the manifest
> > -        list references a legacy schema-1 manifest.
> > -
> > -    - **`size`** *int*
> > -
> > -        The size in bytes of the object. This field exists so that a client
> > -        will have an expected size for the content before validating. If the
> > -        length of the retrieved content does not match the specified length,
> > -        the content should not be trusted.
> > -
> > -    - **`digest`** *string*
> > +  Fields of each object in the manifests list are:
>
> This wasn't really a statement about the manifests array itself, its just
> that this is where I was in the doc when it dawned on me that downloading
> this manifest json by itself feels like the main usecase being promoted -
> so I started to wonder about the "big bundle" case. So, should we consider
> creating a blob (tar) containing this manifest and all referenced images as
> files in there?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/opencontainers/image-spec/pull/12/files/8046ab707b00e89eb9d48b0634ecf23a7c57ef27#r59564875>
>